Add some missing items, weapons, mods, etc.

helixhamin opened the pull request:
Giving up on trying to complete the list this time around, before something comes in and messes up what I have done so far...

<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

helixhamin committed:
Add some missing items, weapons, mods, etc.
Giving up on trying to complete the list this time around, before something comes in and messes up what I have done so far...

Also, added require("scripts/globals/items") to the item luas, and added the new use items to items.lua.

Update: Fixed the item luas to change the item id to the item name after adding to items.lua.

claywar commented on scripts/globals/items/beryllium_bolt_quiver.lua:
```diff
@@ -0,0 +1,22 @@
+end
+
+item_object.onItemUse = function(target)
+    target:addItem(22285, 99)
```
For item references, can you please add the appropriate line to `scripts/globals/items.lua`, and then add the require and reference that table in these scripts?

TeoTwawki replied:
and also tick the boxes on the pr template..

helixhamin replied:
Sorry, I keep forgetting about those boxes.
I will add them Claywar. Sorry, just noticed.

helixhamin commented:
@claywar @TeoTwawki Okay, changes made. Sorry that it took some time.

Era-Lusiphur commented:
Quick glance over. Looks like you've added the items to the item global correctly, but aren't referring to the global in the other scripts.

Would have to include the global in the script files at the top of each
`require("scripts/globals/items")`

and then change where you add the item so that
`5076`
would end up being
`xi.items.SCROLL_OF_FOE_SIRVENTE`

Barring any other requests from the others, then you should be good.

Xaver-DaRed commented on scripts/globals/items/beryllium_bolt_quiver.lua:
```diff
@@ -0,0 +1,23 @@
+end
+
+item_object.onItemUse = function(target)
+    target:addItem(22285, 99)
```
Adding to the comment, you added the wrong items to the table.
Not a problem IMO, they don't hurt there.

`target:addItem(22285, 99)`
In this line the item id is 22285 (Wich are beryllium bolts) with 99 being the quantity.
So in the items.lua file, you should have added those. For example...
`BERYLLIUM_BOLT = 22285,`

And then this line
`target:addItem(22285, 99)`
should be changed to...
`target:addItem(xi.items.BERYLLIUM_BOLT, 99)`

This is partly my fault for not being clear enough in discord. My apologies.

helixhamin commented:
> Quick glance over. Looks like you've added the items to the item global correctly, but aren't referring to the global in the other scripts.
> 
> Would have to include the global in the script files at the top of each `require("scripts/globals/items")`

Those are added to all scripts in global/items/x.lua, fixed that with the last submission.

Era-Lusiphur commented:
> 
> 
> > Quick glance over. Looks like you've added the items to the item global correctly, but aren't referring to the global in the other scripts.
> > Would have to include the global in the script files at the top of each `require("scripts/globals/items")`
> 
> Those are added to all scripts in global/items/x.lua, fixed that with the last submission.

Ah, you're right. Was a quick glance, my mistake. :)

helixhamin commented:
@Era-Lusiphur No worries, thought maybe I missed one, even though I double checked it.
@Xaver-DaRed All good, easy to update, and all updated. Had to double check to make sure I applied them in both items.lua and in the item scripts.

TeoTwawki commented on sql/item_weapon.sql:
```diff
@@ -3556,10 +3556,15 @@ INSERT INTO `item_weapon` VALUES (20561,'carnwenhan',2,0,242,242,228,1,1,186,92,
 INSERT INTO `item_weapon` VALUES (20562,'carnwenhan',2,0,242,242,228,1,1,186,92,0);
 INSERT INTO `item_weapon` VALUES (20563,'twashtar',2,0,242,242,228,1,1,176,89,0);
 INSERT INTO `item_weapon` VALUES (20564,'twashtar',2,0,242,242,228,1,1,176,89,0);
+INSERT INTO `item_weapon` VALUES (20568,'wind_knife_+1',2,0,0,0,0,1,1,189,2,0);     -- Still needs added effect and enchantment
```
```sql
-- Wind Knife +1
INSERT INTO `item_mods` VALUES (20568,431,1);   -- Additional effect: Wind Damage (Type DAMAGE)
INSERT INTO `item_mods` VALUES (20568,499,3);   -- Additional effect animation (subEffect)
INSERT INTO `item_mods` VALUES (20568,500,2);   -- Additional effect damage before resists
INSERT INTO `item_mods` VALUES (20568,501,10);  -- Additional effect Chance 10%
INSERT INTO `item_mods` VALUES (20568,950,8);   -- Additional effect element Wind
```
That's how you add additional effect damage and the format I use.
3 is the subeffect value for wind damage animations, you can find these in the battleentity enum [here](https://github.com/LandSandBoat/server/blob/base/src/map/entities/battleentity.h#L291)

Proc rate of 10% is a guesstimate on my part. Most effects are 5 or 10 rarely an item like sirroco kukri is 100%

Damage should be the MOST the effect can hit for, resists will lower it. 

In nearly all cases for melee weapons NOTHING will increase the damage. Not your int, not your magic attack. Nothing. Ranged attacks DO use INT or MND (MND for light element dmg, INT for all other elements). In very rare cases items exist that use modifiers - we'll handle those if and when we hit them, so don't worry about those for now.


For the enchantment:
```lua
require("scripts/globals/spell_data")

player:castSpell(xi.magic.spell.AERO, target)
```
Could work, but I'm not sure if items targeting mobs is working.

I suggest testing, if it doesn't work leave the enchant commented out.

helixhamin replied:
Sure, noted. For now though, I am trying to stick to completing the others first. There are still a few missing items, and many missing item_mods. And, well, you know me, it will take forever of asking and clarifying to get these right. So, for now, the comment I felt was just better so me or someone else later knows that it is not finished yet, and needs to be done, without taking a few weeks to try to get it done. (And yes, it would take me that long, as you know, I am not that great at those parts xD).
Later, I will make other submissions, and am trying to also figure out everything that is wrong/missing by going over each and every item (my previous commit). But, for this run, this is simply adding in some missing items and added some mods to some items that are there, but are incomplete (many lvl 99, ilvl 119 seem to be missing the item_weapon and item_mods).

helixhamin replied:
I will see what I can do on this. No guarantees though...

helixhamin replied:
Took the last option (per discussion on discord), and just left the whole script as a todo, other than headers and requirements.

helixhamin commented:
Sorry Teo, taking out the script I made for the wind knife +1, since it seems to break something.

TeoTwawki commented:
> Sorry Teo, taking out the script I made for the wind knife +1, since it seems to break something.

Because you didn't construct the empty script correctly. I'll push it myself

TeoTwawki committed:
Add placeholder script for wind knife +1
This is untested, sop the command is commented out and a message left in case some goofball uncomments the item_usable.sql row without confirming its working.

TeoTwawki committed:
Add correct model ID for Wind Knife and Wind Knife +1

TeoTwawki commented:
I have added 2 commits to your pr. please don't be adding anymore sql rows that use placeholder values for the unknowns, that just leads to new opened issues with us by the next person who doesn't read your comment and these are easily findable there's a whole discord devoted to requests that can be tapped.

TeoTwawki committed:
Add missing model IDs for Esikuva, Norgish dagger, Infiltrator...

TeoTwawki committed:
Correct parameter Teo forgot the 1st time - per function definitions this is legit.
https://github.com/LandSandBoat/server/blob/53dfa313d297abd68fb9cb070b286f94a6d237e3/src/map/lua/luautils.cpp#L2380

claywar committed:
Merge branch 'base' into base
